src/Solvers/conjugate_gradient_poisson_solver.jl

```diff
@@ -34,33 +34,47 @@ end
     @inbounds ∇²ϕ[i, j, k] = ∇²ᶜᶜᶜ(i, j, k, grid, ϕ)
 end
 
-function compute_laplacian!(∇²ϕ, ϕ)
+struct RegularizedLaplacian{D}
+    δ :: D
+end
+
+function (L::RegularizedLaplacian)(Lϕ, ϕ)
     grid = ϕ.grid
     arch = architecture(grid)
     fill_halo_regions!(ϕ)
-    launch!(arch, grid, :xyz, laplacian!, ∇²ϕ, grid, ϕ)
+    launch!(arch, grid, :xyz, laplacian!, Lϕ, grid, ϕ)
+
+    if !isnothing(L.δ)
+        # Add regularization
+        ϕ̄ = mean(ϕ)
+        parent(Lϕ) .+= L.δ * ϕ̄
+    end
+
     return nothing
 end
 
 struct DefaultPreconditioner end
 
 function ConjugateGradientPoissonSolver(grid;
+                                        regularization = nothing,
                                         preconditioner = DefaultPreconditioner(),
                                         reltol = sqrt(eps(grid)),
                                         abstol = sqrt(eps(grid)),
                                         kw...)
 
     if preconditioner isa DefaultPreconditioner # try to make a useful default
-        if grid isa ImmersedBoundaryGrid && grid.underlying_grid isa GridWithFFTSolver
-            preconditioner = fft_poisson_solver(grid.underlying_grid)
+        if has_fft_poisson_solver(grid)
+            preconditioner = fft_poisson_solver(grid)
         else
-            preconditioner = DiagonallyDominantPreconditioner()
+            preconditioner = AsymptoticPoissonPreconditioner()
         end
     end
 
     rhs = CenterField(grid)
+    operator = RegularizedLaplacian(regularization)
```

Comment:
It might be a stupid question. It seems to me that `regularization` and `δ` is set to a positive number, 1 for example in the demo. However, the Laplacian operator is actually negative semi-definite \(although our previous conversations were about positive semi-definite\), and thus `regularization` \(or say`δ`\) should be negative as well. Am I missing something?

Reply:
Hmm, right... I played around this and found that the solver "doesn't blow up" only if we use positive `regularization`. But this may actually be a clue that there's a bug somewhere.

Reply:
I'm now convinced that there is a bug or some conceptual error given that this doesn't work with $\delta < 0$